File: core/src/main/java/com/sfl/coolmonkey/escommons/core/component/MappingsComponent.java
package com.sfl.coolmonkey.escommons.core.component;

import javax.annotation.Nonnull;

/**
 * Created by Arthur Asatryan.
 * Date: 7/10/17
 * Time: 6:11 PM
 */
public interface MappingsComponent {
    /**
     * Read mappings for index name and type.
     *
     * @param aliasName the index name
     * @param type      the type
     * @return the mappings
     */
    @Nonnull
    String readMappings(@Nonnull final String aliasName, @Nonnull final String type);
}

File: core/src/main/kotlin/org/biacode/escommons/core/component/MappingsComponent.kt
package org.biacode.escommons.core.component

/**
 * Created by Arthur Asatryan.
 * Date: 7/10/17
 * Time: 6:11 PM
 */
interface MappingsComponent {
    /**
     * Read mappings for index name and type.
     *
     * @param aliasName the index name
     * @param type      the type
     * @return the mappings
     */
    fun readMappings(aliasName: String, type: String): String
}
